storm-data-contracts/src/main/java/com/forter/contracts/BaseContractsBoltExecutor.java

```diff
@@ -230,4 +224,31 @@ public ValidatedContract apply(Object input) {
             return ContractValidator.instance().validate(input);
         }
     }
+
+    public static class ContractViolationReportedFailedException extends ReportedFailedException {
+        private final List<ContractViolationException> violations;
+
+        public ContractViolationReportedFailedException(List<ContractViolationException> violations, String id) {
+            super("There were some contract violations in '" + id + "' bolt." + getViolationsString(violations));
+            this.violations = violations;
+        }
+
+        private static String getViolationsString(List<ContractViolationException> violations) {
+            StringBuilder sb = new StringBuilder();
+
+            if (!violations.isEmpty()) {
+                sb.append(" ");
+                sb.append("Detected violations were:").append("\n");
+                for (ContractViolationException cve : violations) {
+                    sb.append(" - ").append(cve.getMessage()).append("\n");
+                }
+            }
+
+            return sb.toString();
+        }
+
+        public List<ContractViolationException> getContractViolations() {
+            return violations;
+        }
+    }
```

Comment:
@Override
    public Throwable fillInStackTrace()
    {
        return this;
    }